code/modules/economy/worth_cash.dm

```diff
@@ -164,36 +164,148 @@
 	loaded_worth += amt
 	if(loaded_worth < 0)
 		loaded_worth = 0
 
-/obj/item/charge_card/examine(mob/user, distance)
+/obj/item/charge_stick/examine(mob/user, distance)
 	. = ..(user)
 	if(distance <= 2 || user == loc)
-		to_chat(user, SPAN_NOTICE("<b>Owner:</b> [owner_name]."))
-		var/decl/currency/cur = decls_repository.get_decl(currency)
-		to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
+		var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_WARNING("\The [src] is locked."))
+		else
+			to_chat(user, SPAN_NOTICE("<b>Id:</b> [id]."))
+			var/decl/currency/cur = decls_repository.get_decl(currency)
+			to_chat(user, SPAN_NOTICE("<b>[capitalize(cur.name)]</b> remaining: [Floor(loaded_worth / cur.absolute_value)]."))
 
-/obj/item/charge_card/get_base_value()
+/obj/item/charge_stick/get_base_value()
 	. = holographic ? 0 : loaded_worth
 
+/obj/item/charge_stick/attackby(var/obj/item/W, var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+
+	if(istype(W, /obj/item/charge_stick))
+		var/obj/item/charge_stick/sender = W
+		var/datum/extension/lockable/W_lock = get_extension(W, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds to a locked [src]."))
+			return TRUE
+		if(W_lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds from a locked [W]."))
+			return TRUE
+		if(sender.currency != currency)
+			to_chat(user, SPAN_WARNING("Mismatched currency detected. Unable to transfer."))
+			return TRUE
+		var/amount = input(user, "How much of [sender.loaded_worth] do you want to transfer?", "[sender] transfer", "0") as null|num
+		if(!amount)
+			return TRUE
+		if(amount < 0 || amount > sender.loaded_worth)
+			to_chat(user, SPAN_WARNING("Enter a valid number between 1 and [sender.loaded_worth] to transfer."))
+			return TRUE
+		sender.loaded_worth -= amount
+		loaded_worth += amount
+		sender.update_icon()
+		update_icon()
+		var/decl/currency/cur = decls_repository.get_decl(currency)
+		to_chat(user, SPAN_NOTICE("Completed transfer of [amount] [cur.name]."))
+		return TRUE
+	
+	if(lock.attackby(W, user))
+		return TRUE
+	return ..()
+
+/obj/item/charge_stick/emag_act(var/remaining_charges, var/mob/user, var/feedback)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	.= lock.emag_act(remaining_charges, user, feedback)
+
+/obj/item/charge_stick/attack_self(var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	lock.ui_interact(user)		
+
+/obj/item/charge_stick/proc/is_locked()
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	return lock.locked
+
+/obj/item/charge_stick/on_update_icon()
+	. = ..()
+	overlays.Cut()
+	// if(isturf(loc)) // Re-enable this when world icons are functional.
```

Comment:
Re-enable this when world icons are functional

Reply:
Also since you call ..() you don't need to cut overlays, item's update icon does that.
Also maybe just do 
```
if(get_world_inventory_state() == ICON_STATE_WORLD)
  return
```